src/types/datetime.rs

```diff
@@ -279,7 +252,7 @@ impl PyTime {
         minute: u8,
         second: u8,
         microsecond: u32,
-        tzinfo: Option<&PyObject>,
+        tzinfo: Option<&PyTzInfo>,
```

Comment:
Wouldn't this be a breaking change? It may probably affect pyo3-chrono crate. Maybe need to ping the author if this is indeed the case.

Reply:
Yes, and we'll release this as part of the 0.14 release. I've notified this in the CHANGELOG, and we are still pre-1.0, after all. If it turns out to be a big problem we can revert it later.

Reply:
I can't quite remember why I went with `&PyObject` instead of `&PyTzInfo` here originally. It's entirely possible that I misunderstood how inheritance in Rust and Python can play together well, The `datetime.tzinfo` class is an abstract base class (but I think one that was created before Python supported either abstract base classes or Protocols).

In theory it should be a trait, but in practice `datetime` enforces that it is an actual subclass of `datetime.tzinfo` and I don't know if I'll be able to change that any time soon. Presumably it is safe to change this from `PyTzInfo` back to `PyObject` after the 1.0 release, since all `PyTzInfo` objects are `PyObject`s but not vice versa?

Reply:
Ah, I remember why it has to be `&PyObject` now — `tzinfo` can also be `None` (and in fact probably _is_ `None` in the common case, since `datetime.timezone` is not a very good time zone representation but datetimes are useful anyway).

I'm actually fairly surprised that this doesn't break any tests.

Reply:
Ah, sorry to keep going back and forth here, but I forgot that this is `Option<&PyTzInfo>`, and that the `None` is handled by the `None` case of `Option`, so I'm back to not remembering why this was `PyObject` instead of `PyTzInfo`.